src/main/scala/viper/gobra/Gobra.scala

```diff
@@ -42,38 +42,88 @@ object GoVerifier {
   }
 }
 
-trait GoVerifier {
+trait GoVerifier extends StrictLogging {
+
+  type Warning = String
 
   def name: String = {
     this.getClass.getSimpleName
   }
 
-  def verify(config: Config)(executor: GobraExecutionContext): Future[VerifierResult] = {
-    verify(config.inputs, config)(executor)
+  def verifyAllPackages(config: Config)(executor: GobraExecutionContext): Vector[VerifierError] = {
+    val statsCollector = StatsCollector(config.reporter)
+    var warningCount: Int = 0
+    var allErrors: Vector[VerifierError] = Vector()
+
+    // write report to file on shutdown
+    Runtime.getRuntime.addShutdownHook(new Thread() {
+      override def run(): Unit = {
+        val statsFile = config.gobraDirectory.resolve("stats.json").toFile
+        logger.info("Writing report to " + statsFile.getPath)
+        statsCollector.writeJsonReportToFile(statsFile)
+      }
+    })
+
+    config.inputPackageMap.foreach({ case (pkg, inputs) =>
+      val taskName = pkg.path + " - " + pkg.name
+
+      logger.info("Verifying Package " + taskName)
+      val resultFuture = verify(config.copy(inputs = inputs, reporter = statsCollector), Some(taskName))(executor)
+      val result = Await.result(resultFuture, Duration.Inf)
```

Comment:
I think that this isn't possible with futures, since they start once created, which means they would still be executed in parallel or am I wrong?

Reply:
But I packed the reporting step into the `Future`

Reply:
You can use e.g. `foldLeft` to iterate over the collection and extend the change of futures by waiting for the result of the previous future before verifying the current package (use `.map()` to do so)

Reply:
exactly. That's how it's done in the chopper as well

Reply:
Okay, I see what you mean now, but I won't be able to timeout these futures on a package level(see updated code) because I don't think there is a different solution for timeouts than a blocking call like `Await.result`

Reply:
This is true. However, you're currently using `Duration.Inf` anyway

Reply:
No, GitHub unfortunatelly doesn't update this view, the current version looks like this:
```
      try {
        Await.result(future, config.packageTimeout)
      } catch {
        case _: TimeoutException => logger.error("The verification of package " + pkgId + " got terminated after " + config.packageTimeout.toString)
      }
```